Code/GraphMol/FileParsers/SmilesMolSupplier.cpp

```diff
@@ -41,6 +41,7 @@ SmilesMolSupplier::SmilesMolSupplier(const std::string &fileName,
   if (!tmpStream || (!(*tmpStream)) || (tmpStream->bad())) {
     std::ostringstream errout;
     errout << "Bad input file " << fileName;
+    if (tmpStream) { delete tmpStream; }
```

Comment:
the `if` isn't necessary here; it's safe to call `delete` on `nullptr`

Reply:
Ok.